### DISCUSSION TITLE
How to partially run the workflow

### DISCUSSION THREAD

**Discussion by MG-Programmer** [INTI-CMNB/KiBot#346] [Missing in my workflow]
Hi there,

I want to create a custom workflow with all (for me) relevant outputs based on the --quick-start workflow.
For testing purposes and to avoid creating all the stuff on just a change in e.g. the bom I want to run just a specific action from the workflow.
Is this possible? If so then how can I perform it?

Thanks

**Comment by set-soft** [comment]
Hi @MG-Programmer !

You can do it, but not with the GitHub Action. For an action you'll need the second approach.

You'll need to use a docker image, like in this [workflow](https://github.com/INTI-CMNB/kibot_variants_arduprog/blob/master/.github/workflows/kibot_quick_start.yml)
Then you just need to use `-t` option to indicate which output types do you want. 
I.e. `kibot -t bom -t gerber -t render_3d --quick-start` 
Will generate only `bom`, `gerber` and `render_3d` output types.

But a better option is:
- Run a workflow that uses `--quick-start`, can be a GitHub Action
- Store the generated file/s in an artifact
- Download the artifact
- Rename the file and add it to your repo
- Create a workflow that uses this file
- Ask KiBot to generate only the wanted outputs

Note that you can limit which outputs are generated even for a GitHub Action using `targets`.
You can also have two copies of the generated config file, one with all the outputs and another with the ones you want for quick tests. Then you just need to specify which configuration to use.

In general: Quick Start is supposed to be a mechanism to quickly start using KiBot, not a permanent mechanism. Once you have a configuration file generated by KiBot you just copy the outputs you are interested to a new file. Then you continue doing adjusts to the filtered outputs.
